massa-execution-worker/src/tests/scenarios_mandatories.rs

```diff
@@ -1,173 +1,193 @@
+    initial.insert(get_random_address(), Amount::from_str("129").unwrap());
+    initial.insert(get_random_address(), Amount::from_str("878").unwrap());
+    let (cfg, tempfile) = LedgerConfig::sample(&initial);
+    Ok((Arc::new(RwLock::new(FinalLedger::new(cfg)?)), tempfile))
 }
 
-#[tokio::test]
+#[test]
 #[serial]
-async fn test_execution_basic() {
-    let (_config_file_keepalive, settings) = get_sample_settings();
-    assert!(start_controller(settings, None).await.is_ok());
+fn test_execution_basic() {
+    let (sample_ledger, _keep) = get_sample_ledger().unwrap();
+    let (_, _) = start_execution_worker(ExecutionConfig::default(), sample_ledger);
 }
 
-#[tokio::test]
+#[test]
 #[serial]
-async fn test_execution_shutdown() {
-    let (_config_file_keepalive, settings) = get_sample_settings();
-    let (_command_sender, _event_receiver, manager) = start_controller(settings, None)
-        .await
-        .expect("Failed to start execution.");
-    manager.stop().await.expect("Failed to stop execution.");
+fn test_execution_shutdown() {
+    let (sample_ledger, _keep) = get_sample_ledger().unwrap();
+    let (mut manager, _) = start_execution_worker(ExecutionConfig::default(), sample_ledger);
+    manager.stop()
 }
 
-#[tokio::test]
+#[test]
 #[serial]
-async fn test_sending_command() {
-    let (_config_file_keepalive, settings) = get_sample_settings();
-    let (command_sender, _event_receiver, manager) = start_controller(settings, None)
-        .await
-        .expect("Failed to start execution.");
-    command_sender
-        .update_blockclique(Default::default(), Default::default())
-        .await
-        .expect("Failed to send command");
-    manager.stop().await.expect("Failed to stop execution.");
+fn test_sending_command() {
+    let (sample_ledger, _keep) = get_sample_ledger().unwrap();
+    let (mut manager, controller) =
+        start_execution_worker(ExecutionConfig::default(), sample_ledger);
+    controller.update_blockclique_status(Default::default(), Default::default());
+    manager.stop()
 }
 
-#[tokio::test]
+#[test]
 #[serial]
-async fn test_sending_read_only_execution_command() {
-    let (_config_file_keepalive, settings) = get_sample_settings();
-    let (command_sender, _event_receiver, manager) = start_controller(settings, None)
-        .await
-        .expect("Failed to start execution.");
-    command_sender
-        .execute_read_only_request(
-            Default::default(),
-            Default::default(),
-            Default::default(),
-            Default::default(),
-        )
-        .await
-        .expect("Failed to send command");
-    manager.stop().await.expect("Failed to stop execution.");
+fn test_sending_read_only_execution_command() {
+    let (sample_ledger, _keep) = get_sample_ledger().unwrap();
+    let (mut manager, controller) =
+        start_execution_worker(ExecutionConfig::default(), sample_ledger);
+    controller
+        .execute_readonly_request(ReadOnlyExecutionRequest {
```

Comment:
Is there a way to check that it has been executed ?

Reply:
We can look at the events as in `generate_events` for that yes